pymc3/model.py

```diff
@@ -1074,17 +1084,24 @@ def register_rv(self, rv_var, name, data=None, total_size=None, dims=None):
             # In all other cases, the role of the value variable is taken by
             # observed data. That's why value variables are only referenced in
             # this branch of the conditional.
-            value_var = rv_var.clone()
-            value_var.name = rv_var.name
-            rv_var.tag.value_var = value_var
-
             self.free_RVs.append(rv_var)
+            value_var = rv_var.clone()
 
-            transform = rv_var.tag.transform
-            value_var.tag.transform = None
-
+            transform = transform or logp_transform(rv_var.owner.op, rv_var.owner.inputs)
             if transform is not None:
-                self.deterministics.append(rv_var)
+                value_var.tag.transform = transform
+                value_var.name = f"{rv_var.name}_{transform.name}"
```

Comment:
Any reason to drop the dunder at the end of the transformed variable name?

Reply:
No, no reason.  Why was it there in the first place?

Reply:
Preceeded me. My guess would be to avoid name collisions on the user side.